src/amp-story-embed.js

```diff
@@ -150,7 +247,18 @@ self.onload = () => {
     const embed = embeds[i];
     const embedImpl = new AmpStoryEmbed(self, embed);
     embedImpl.buildCallback();
-    // TODO(Enriqe): add intersection observer that triggers layoutCallback().
-    embedImpl.layoutCallback();
+
+    const intersectingCallback = entries => {
+      entries.forEach(entry => {
+        if (!entry.isIntersecting) {
+          return;
+        }
+        embedImpl.layoutCallback();
+      });
+    };
+    const observer = new IntersectionObserver(intersectingCallback, {
+      threshold: 0.5,
```

Comment:
I think we might want to consider lowering this threshold considerably.  In fact, I think we probably want to preload `n` viewports in advance.

Reply:
+1, we should do 2 viewports to match the AMP behavior. We should also have a fallback for when `IntersectionObserver` isn't supported.